### DISCUSSION TITLE
Setting initial conditions in restart files

### DISCUSSION THREAD

**Discussion by liangwang0734** [PrincetonUniversity/athena#616] [Q&A] [answered]
Hi Athena++ community, how I can set the state variable values in a restart file using an external data source (with B field also on Yee mesh)?

- My case of application is not really external: We have an Athena++ simulation with 4x4x4=64 meshblocks per processor (after two uniform refinements) and found it seems to be scale worse than a simulation with the same global mesh size but only 1 large meshblock per processor. Thus we want to dump the data from the former simulation to a restart file compatible with the latter simulation.

**Comment by msbc** [comment]
What you want to do here is to make sure to dump out data in double precision (`-h5double`; configuration flag) and dump both the conservative variables (`cons`) and face centered B-fields (`b`). Then you basically just use the code from https://github.com/PrincetonUniversity/athena/blob/master/src/pgen/from_array.cpp in your pgen to read those hdf5 files and set the corresponding mesh block data.

**Comment by liangwang0734** [comment]
Thank you @msbc , I will try that.

**Comment by msbc** [comment, marked as answer]
@liangwang0734 I just realized that I forgot to mention that the `-h5double` flag is a configuration time option, so you'll have to recompile your athena executable.
